Feature: enable multi-k calculation for CUDA version of module_gint (#4839)
deepmodeling/abacus-develop: docs/advanced/input_files/input-main.md

  - cpu: for CPUs via Intel, AMD, or Other supported CPU devices
  - gpu: for GPUs via CUDA or ROCm.

>   Known limitations: If using the pw basis, the ks_solver must be cg/bpcg/dav to support `gpu` acceleration. lcao_in_pw currently does not support `gpu`.

Reviewer: ```suggestion
  Known limitations: `ks_solver` must also be set to the algorithms supported. lcao_in_pw currently does not support `gpu`.
```
